### DISCUSSION TITLE
prevent MultiSelectionPrompt from clearing console

### DISCUSSION THREAD

**Discussion by jwilburn** [spectreconsole/spectre.console#624] [Q&A]
Is there a setting to prevent `MultiSelectionPrompt` from clearing the console after a user accepts the selection(s)? it looks like the other prompt types don't clear the console after data input.

**Comment by nils-a** [comment]
@jwilburn can you show an example of what you mean? None of the prompts clear the console. 

However, `SelectionPrompt` and `MultiSelectionPrompt` are removed from the console, after a selection has been made. 

So, in theory, if you have a large amount of selections and set the `PageSize` to a large value, all content of the console will be pushed "up" to make room for the `SelectionPrompt`. If you then select an item, the prompt will be removed and the cursor will appear in the first row of the console, making it look like the console has been cleared.

**Reply by jwilburn** [reply]
@nils-a apologies for the confusion, I probably used the wrong verbiage when trying to explain the issue I was having.

You mentioned the following in your previous post ...
>However, SelectionPrompt and MultiSelectionPrompt are removed from the console, after a selection has been made.

Is it possible to not remove them from the console after a selection is made? if not, no biggie, just wanted to make sure I wasn't missing a flag somewhere.

**Reply by nils-a** [reply]
> Is it possible to not remove them from the console after a selection is made?

No, this is currently not possible.

**Reply by ku3mich** [reply]
@nils-a can you point me to detailed explanation 'how to properly remove a IRederable from console using LiveView' how this SelectionPrompt does?
